var/spack/repos/builtin/packages/node-js/package.py

```diff
@@ -36,6 +36,7 @@ class NodeJs(Package):
     variant('icu4c', default=False, description='Build with support for all locales instead of just English')
     variant('openssl', default=True,  description='Build with Spacks OpenSSL instead of the bundled version')
     variant('zlib', default=True,  description='Build with Spacks zlib instead of the bundled version')
+    variant('without-npm', default=True, description='Build without npm dependency (to be installed explicitly)')
```

Comment:
Can you explain why this variant is necessary? Why can't a package depend on both `node-js` and `npm`?

Reply:
Yes, in almost all cases, someone would want to use both `node-js` and `npm`. In the prev package, `--without-npm` was being passed to configure step, with the following comment.
```
# Note: npm is updated more regularly than node.js, so we build
# the package instead of using the bundled version
```

However, i find this wrong, particularly because `npm` failed for me due to some very deep dependency incompatibility. Here, I propose putting `--without-npm` option as a variant, so people can choose. e.g., it worked with me to install `node-js` with `npm`.